src/hotspot/share/gc/g1/g1Policy.cpp

```diff
@@ -523,7 +525,11 @@ void G1Policy::revise_young_list_target_length_if_necessary(size_t rs_length) {
     // add 10% to avoid having to recalculate often
     size_t rs_length_prediction = rs_length * 1100 / 1000;
     update_rs_length_prediction(rs_length_prediction);
-    update_young_length_bounds(rs_length_prediction);
+
+    G1DirtyCardQueueSet& dcqs = G1BarrierSet::dirty_card_queue_set();
+    // We have no measure of the number of cards in the thread buffers, assume
+    // these are very few compared to the ones in the DCQS.
```

Comment:
I think it's really that the #cards in thread buffers is very small compared to the sum of the other two sources.

Reply:
Note that this comment does not say that what is assumed here is actually valid in all cases, but for now it suffices I guess.